q2_composition/tests/test_ancom.py

```diff
@@ -155,6 +155,26 @@ def test_ancom_no_tables(self):
             f = fh.read()
             self.assertTrue('No significant features found' in f)
 
+    def test_ancom_zero_division(self):
+        t = pd.DataFrame([[10, 0], [11, 0], [12, 0], [13, 0],
+                          [1000, 10], [1000, 10]],
+                         index=['S1', 'S2', 'S3', 'S4', 'S5', 'S6'],
+                         columns=['O1', 'O2'])
+        c = qiime2.CategoricalMetadataColumn(
+            pd.Series(['0', '0', '1', '1', '2', '2'], name='n',
+                      index=pd.Index(['S1', 'S2', 'S3', 'S4', 'S5', 'S6'],
+                                     name='id'))
+        )
+
+        ancom(output_dir=self.temp_dir.name, table=t+1, metadata=c,
+              transform_function='log')
+
+        with open(os.path.join(self.temp_dir.name, 'index.html')) as fh:
+            f = fh.read()
+            self.assertFalse('Infinity' in f)
```

Comment:
Will `'Infinity' in f` ever evaluate to False?  I didn't see 'Infinity' anywhere in the assets.  Just wondering how it would get in there.  Would that be javascript's doing?

Reply:
Yeah, great question - this has to do with serializing the data to JSON - prior to this PR, yes, its totally possible (if you run this test case `test_ancom_zero_division` manually you will wind up with `Infinity` in the rendered JSON). Ideally it is impossible for this to happen now, given the changes proposed here.